### ISSUE TITLE
Couldn't find all resumable slots by key/index during replaying.

### ISSUE THREAD

**Issue by jahirfiquitiva** [vercel/next.js#63210]
### Link to the code that reproduces this issue

https://github.com/jahirfiquitiva/jahir.dev

### To Reproduce

1. Deploy on Vercel.
2. Visit the deployment URL
3. Visit a blog post (example: https://website-bmqg5pjuv-jahirfiquitiva.vercel.app/blog/how-tailwind-grew-on-me)
4. Refresh/reload the page (cmd+R)

### Current vs. Expected behavior

**Current:**

The website shows `not found` after reloading, and this error is logged in Vercel

```gradle
Error: Couldn't find all resumable slots by key/index during replaying. The tree doesn't match so React will fallback to client rendering.
    at nW (/var/task/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:70706)
    at nz (/var/task/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:69570)
    at nG (/var/task/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:73218)
    at nW (/var/task/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:71383)
    at nz (/var/task/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:69570)
    at nH (/var/task/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:63588)
    at nz (/var/task/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:67383)
    at nB (/var/task/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:51051)
    at nH (/var/task/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:51758)
    at nH (/var/task/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:66179) {
  digest: '4172697567'
}
```

https://github.com/vercel/next.js/assets/10360816/e24018b4-76e3-4b68-99b4-a54d952e3578

**Expected:**

The website keeps loading the site normally

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.3.0: Wed Dec 20 21:30:44 PST 2023; root:xnu-10002.81.5~7/RELEASE_ARM64_T6000
  Available memory (MB): 16384
  Available CPU cores: 10
Binaries:
  Node: 20.11.0
  npm: 10.2.4
  Yarn: 4.1.0
  pnpm: 8.15.4
Relevant Packages:
  next: 14.2.0-canary.18 // Latest available version is detected (14.2.0-canary.18).
  eslint-config-next: 14.1.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Which stage(s) are affected? (Select all that apply)

Vercel (Deployed)

**Comment by wyattjoh** [comment]
This should get resolved by an upcoming fix over here https://github.com/vercel/vercel/pull/11259, once this is merged and we release a new version, I'll update here for you to try a redeploy!

**Comment by jahirfiquitiva** [comment]
@wyattjoh  Thanks for letting me know ... looking forward to it! 🙌

**Comment by mwskwong** [comment]
> This should get resolved by an upcoming fix over here [vercel/vercel#11259](https://github.com/vercel/vercel/pull/11259), once this is merged and we release a new version, I'll update here for you to try a redeploy!

@wyattjoh Just wanna follow up on this. May I know if there is an ETA for this?

**Comment by jahirfiquitiva** [comment]
@mwskwong I think Vercel is now using v `33.6.1` which seems to include the fix, give it a try. It is working for me now

**Comment by mwskwong** [comment]
> @mwskwong I think Vercel is now using v `33.6.1` which seems to include the fix, give it a try. It is working for me now
> 
> Double check this on Vercel 👇
> 
> ![Shot 2024-03-22 at 12 48 15@2x](https://private-user-images.githubusercontent.com/10360816/316120680-5b979861-295b-4c31-9814-b8a44c7408fe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEzMzUzODIsIm5iZiI6MTcxMTMzNTA4MiwicGF0aCI6Ii8xMDM2MDgxNi8zMTYxMjA2ODAtNWI5Nzk4NjEtMjk1Yi00YzMxLTk4MTQtYjhhNDRjNzQwOGZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI1VDAyNTEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFhYzlmY2IwODliZmFkNzA4MDc2NTU5OGZkOGUwMDIwMzliZmRjNDY3MDRlYzJhZWJkOTNkYTk0ZTU1ZjNlYjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DLe9KZkXlRwlsY2bDrM0eMdomnLIhAUcgjsDU2ZHNVw)

Noticed. Thanks.

**Comment by jperezr21** [comment]
Happening to me quite frequently on Vercel:

<img width="596" alt="Image" src="https://github.com/user-attachments/assets/8ad71427-89bd-4d88-b0a5-01b6e2162b8c" />

**Comment by wyattjoh** [comment]
This should get resolved with some of the work on the resume data cache that we're doing. It'll enable the tree to replay more accurately.

**Comment by filiptrivan** [comment]
Any updates?

_Edit by maintainer bot: Comment was **automatically** minimized because it was considered unhelpful. (If you think this was by mistake, let us know). Please only comment if it adds context to the issue. If you want to express that you have the same problem, use the upvote 👍 on the issue description or subscribe to the issue for updates. Thanks!_

**Comment by nikolaf994** [comment]
Seems like it didn't get resolved, I'm still getting this error on nextjs v16.2.4 hosted and deployed on vercel and it is happening the same way as on video in original post. Locally in dev or prod, whatever i do i just can't recreate this issue. 

Error started happening after I moved to "use cache" directive (cacheComponents)

EDIT: Okay, rolling back to next version 16.1.2. sorted out this specific issue for me.

**Comment by laurentlaporte** [comment]
Downgrading to 16.1.2 also fixed it for me... Very annoying

**Comment by LukasBaksys** [comment]
Issue is still there. Maybe anyone found out what causes this and how to work around it?

**Comment by filiptrivan** [comment]
The "16.2.x-only, works on 16.1.2, can't repro locally" reports here are the **bot + streaming-metadata PPR resume** variant.

In minimal mode the platform serves the prerendered shell (built with `serveStreamingMetadata: true`, so the metadata slot is a `<div>`) and only resumes the postponed hole. 16.2.0 broadened the streaming-metadata bot opt-out in `build/templates/app-page.ts` from `isHtmlBot` → `botType`, so Googlebot now resumes with `serveStreamingMetadata: false` and renders `<__next_metadata_boundary__>` where the shell postponed a `<div>` → tree mismatch → client fallback. In 16.1.2 only HTML-limited bots opted out, so Googlebot kept streaming on and matched the shell.

Repros only for recognized bot UAs on dynamic-metadata PPR routes, prod/minimal mode only. Appears addressed by #94630 (open).

**Comment by filiptrivan** [comment]
Sharing the workaround we shipped, in case it helps anyone still stuck. We did not wait for a fix release.

For the bot variant: the main Googlebot is the only UA Next treats as a `dom` bot, so it resumes with `serveStreamingMetadata: false` and hits the slot mismatch. Every other crawler is `html`-limited, gets a blocking render, and its metadata lands in `<head>` fine. So we just make Googlebot take that same `html` path.

In our edge proxy, on dynamic-metadata routes only, we rewrite Googlebot's UA to a token that matches Next's `HTML_LIMITED_BOT_UA_RE` (the `Google-[\w-]+` branch):

```ts
const ua = req.headers.get("user-agent") ?? "";
if (/Googlebot(?!-)|Googlebot$/i.test(ua) && isDynamicMetadataRoute(pathname)) {
  const headers = new Headers(req.headers);
  headers.set("user-agent", ua.replace(/Googlebot/gi, "Google-YourApp-BotHead"));
  return NextResponse.next({ request: { headers } });
}
```

`getBotType()` now returns `html` for Googlebot, so it gets the blocking render and the mismatch is gone. We scope it to dynamic routes so the extra blocking render only happens where it's needed.

Worth flagging for anyone waiting on #94630: that PR stops the mismatch but forces streaming metadata back on, so the tags come back in `<body>` (JS hoists them into `<head>`). If you need them in the raw `<head>` for crawlers, this rewrite is still needed. We guard it with a plain curl as Googlebot asserting `<title>` shows up before `</head>`, since a browser test hides the bug.
